clients/da-vinci-client/src/main/java/com/linkedin/davinci/stats/DIVStatsReporter.java

```diff
@@ -21,53 +21,46 @@ public DIVStatsReporter(MetricsRepository metricsRepository, String storeName) {
     registerSensor(
-        new DIVStatsCounter(
-            "potentially_lossy_leader_offset_rewind_count",
+        new DIVStatsGauge(
             this,
-            () -> (double) getStats().getPotentiallyLossyLeaderOffsetRewindCount()));
+            () -> (double) getStats().getPotentiallyLossyLeaderOffsetRewindCount(),
+            "potentially_lossy_leader_offset_rewind_count"));
     registerSensor(
-        new DIVStatsCounter(
-            "leader_producer_failure_count",
-            this,
-            () -> (double) getStats().getLeaderProducerFailure()));
+        new DIVStatsGauge(this, () -> (double) getStats().getLeaderProducerFailure(), "leader_producer_failure_count"));
     registerSensor(
-        new DIVStatsCounter(
-            "benign_leader_producer_failure_count",
+        new DIVStatsGauge(
             this,
-            () -> (double) getStats().getBenignLeaderProducerFailure()));
+            () -> (double) getStats().getBenignLeaderProducerFailure(),
+            "benign_leader_producer_failure_count"));
 
     // This prevents user store system store to register latency related DIV metric sensors.
     if (!VeniceSystemStoreUtils.isUserSystemStore(storeName)) {
-      registerLatencySensor("producer_to_source_broker", DIVStats::getProducerSourceBrokerLatencySensor);
-      registerLatencySensor("source_broker_to_leader_consumer", DIVStats::getSourceBrokerLeaderConsumerLatencySensor);
-      registerLatencySensor("producer_to_local_broker", DIVStats::getProducerLocalBrokerLatencySensor);
-      registerLatencySensor("local_broker_to_follower_consumer", DIVStats::getLocalBrokerFollowerConsumerLatencySensor);
-      registerLatencySensor("leader_producer_completion", DIVStats::getLeaderProducerCompletionLatencySensor);
       registerLatencySensor("leader_div_completion", DIVStats::getLeaderDIVCompletionLatencySensor);
+      registerLatencySensor("drainer_div_completion", DIVStats::getDrainerDIVCompletionLatencySensor);
     }
   }
 
   protected void registerLatencySensor(
       String sensorBaseName,
       Function<DIVStats, WritePathLatencySensor> sensorFunction) {
     registerSensor(
-        new DIVStatsCounter(sensorBaseName + "_latency_avg_ms", this, () -> sensorFunction.apply(getStats()).getAvg()));
+        new DIVStatsGauge(this, () -> sensorFunction.apply(getStats()).getAvg(), sensorBaseName + "_latency_avg_ms"));
     registerSensor(
-        new DIVStatsCounter(sensorBaseName + "_latency_max_ms", this, () -> sensorFunction.apply(getStats()).getMax()));
+        new DIVStatsGauge(this, () -> sensorFunction.apply(getStats()).getMax(), sensorBaseName + "_latency_max_ms"));
   }
 
-  protected static class DIVStatsCounter extends AsyncGauge {
-    DIVStatsCounter(String metricName, DIVStatsReporter reporter, DoubleSupplier supplier) {
+  protected static class DIVStatsGauge extends AsyncGauge {
+    DIVStatsGauge(DIVStatsReporter reporter, DoubleSupplier supplier, String metricName) {
```

Comment:
Is this change useful? I believe it will cause a rename of metrics, right? While we might not have dashboards built on these particular metrics, we should still avoid changing metric names unless there is a good reason for it...

And likewise for the parameter order change, is it useful? If the new order is much cleaner, then sure, we can, but if it's equivalent, then IDK if it's worth muddying the understanbility of the git history for it?

Reply:
Yeah, it will change the metric name.
Checking whether these metrics are used in alert dashboard is tedious but doable. Use the metric platform command to list all our alert dashboard and grep for "DIVStatsCounter"

Reply:
I changed it because I want to have some consistency for `DIVStatsGauge` and `IngestionStatsGauge`. They are most frequently used subclasses of `AsyncGauge` and they have a totally reversed parameter order, which looks odd to me. Since I also moved some metrics from `DIVStatsReporter` to `IngestionStatsReporter`, I think it's better to make them consistent

Reply:
And yes I've checked, we haven't set up any alerts on DIV metrics yet so there's no impact on the dashboard.